Buildings/Controls/OBC/CDL/Logical/VariablePulse.mo

```diff
@@ -0,0 +1,340 @@
+within Buildings.Controls.OBC.CDL.Logical;
+block VariablePulse
+  "Generate boolean pulse with the width specified by input"
+
+  parameter Real period(unit="s")
+    "Time for one pulse period";
+  parameter Real chaWidThr=0.01
+    "Minimum input change to re-trigger an update of the output interval. It is the ratio of the value change to the original value"
```

Comment:
Do we really want to make a relative comparison? Wouldn't it be simpler to just check for absolute changes? In that case, the test for smaller than a tolerance can also be removed and the limit can be imposed on the width of the pulse. This will avoid state events caused by `greThr1`

Reply:
Thanks! I removed the relative comparison and avoided using `greThr1`.